netZooPy/panda/panda.py

```diff
@@ -405,30 +414,121 @@ def panda_loop(self, correlation_matrix, motif_matrix, ppi_matrix):
         def t_function(x, y=None):
             '''T function.'''
             if y is None:
-                a_matrix = np.dot(x, x.T)
-                s = np.square(x).sum(axis=1)
-                a_matrix /= np.sqrt(s + s.reshape(-1, 1) - np.abs(a_matrix))
+                a_matrix = xp.dot(x, x.T)
```

Comment:
same here, the `t_function` is used by both gpuPANDA and cpu PANDA, so we don't want to use `xp.dot` in the cpu version. I'd suggest to write a different `t_function` for gpu or have an `if` statement here